clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientAuthenticator.java

```diff
@@ -106,13 +107,21 @@ public void configure(TransportLayer transportLayer, PrincipalBuilder principalB
 
             setSaslState(handshakeRequestEnable ? SaslState.SEND_HANDSHAKE_REQUEST : SaslState.INITIAL);
 
-            // determine client principal from subject.
-            if (!subject.getPrincipals().isEmpty()) {
-                Principal clientPrincipal = subject.getPrincipals().iterator().next();
-                this.clientPrincipalName = clientPrincipal.getName();
-            } else {
-                clientPrincipalName = null;
+            // determine client principal from subject for Kerberos. For other mechanisms, the authenticated
+            // principal (username for PLAIN and SCRAM) is used as authorization id.
```

Comment:
I guess clientPrincipalName is only used for Kerberos and not for other mechanisms like PLAIN or SCRAM? If so, perhaps we could make this clearer in the comment.